### ISSUE TITLE
npm audit (December 2021)

### ISSUE THREAD

**Issue by mikeTWC1984** [jhuckaby/Cronicle#460]
I see some new vulnerabilities reported by "npm audit".   "npm audit fix --force" can  clear it, but at this point it breaks cronicle (I see npm applies some changes to source code when fixing mkdirp). Can you do updates below for now

**cronicle and pixl-server-storage**
mkdirp - switch to native fs.mkdir for

**pixl-mail**
switch to latest nodemailer

**Comment by mikeTWC1984** [comment]
for pixl-server-storage also update nearley to the latest version (this will fix several minor warnings)
for mkdirp - may try v1.0.4
Most of the high/critical errors seemed to come from socket.io. Might need to bump to latest version too (4.4.0)

**Comment by mikeTWC1984** [comment]
FYI, mkdirp v1.0.4 won't automatically fix those warnings - 1.x uses promises instead of callbacks. So either need to switch to fs.mkdir or rewrite callbacks to promises.
Also, all pixl-server-XXX packages depends on pixl-server, so need to start updating pixl-server first and then all other components.

**Comment by jhuckaby** [comment]
Understood, thank you!  I'll get started on this.

**Comment by mikeTWC1984** [comment]
Thanks! turns out mkdirp v0.5.5 could work (to clean up warnings), although switching to fs.mkdir sounds like a better idea. I'm testing socket.io v.4.4, so far it doesn't seem to break anything.

**Comment by jhuckaby** [comment]
Yup, I've already ripped out `mkdirp` entirely, and am now using the built-in `fs` impl across all my modules.

Thank you for testing!  I've also upgraded to socket.io 4.4 locally and things seem okay, but I want to try a bunch of multi-server failover scenarios before I call it.

I've checked everything in, but haven't tagged a release yet.

**Comment by jhuckaby** [comment]
Seems stable at work, on a multi-server cluster with multi-master failover and multiple slaves all running jobs simultaneously.  I've tagged a release:

https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.0

I've also started a discussion thread, because this socket.io upgrade is a major breaking change.  Users will run into trouble unless they upgrade their entire clusters at once (socket.io v1.7.3 cannot communicate with v4.4, and visa-versa).

https://github.com/jhuckaby/Cronicle/discussions/461
